doc/api/webstreams.md

```diff
@@ -1457,6 +1457,39 @@ added: v16.7.0
 * Returns: {Promise} Fulfills with an `ArrayBuffer` containing the full
   contents of the stream.
 
+```mjs
+import { arrayBuffer } from 'node:stream/consumers';
+import { Readable } from 'node:stream';
+
+const items = Array.from(
+  {
+    length: 100
+  },
+  () => 'hello world from consumers!'
+);
+
+const readable = Readable.from(JSON.stringify(items));
+const data = await arrayBuffer(readable);
+console.log(`from readable: ${data.byteLength}`);
+```
+
+```cjs
+const { arrayBuffer } = require('node:stream/consumers');
+const { Readable } = require('node:stream');
+
+const items = Array.from(
+  {
+    length: 100
+  },
+  () => 'hello world from consumers!'
+);
+
+const readable = Readable.from(items);
```

Comment:
Actually, for this and the other examples, you could put any data into it. Since they'll eventually become a buffer when you create the readable stream. After that the specific method will only decode that buffer to the specific type. This is not true in the case of JSON because it's supposed to take a byte representation and convert it to JSON

Reply:
You and I know that behind the scenes they will eventually be a buffer. However, you're implying that all users would know that and this might lead them to errors.

I'd explicitly use the datatypes we wanna consume there so when users are trying to consume those specific data types the docs will show them how to use it.

Just in case you would like to see some examples, they're on the tests for the consumers here: https://github.com/nodejs/node/blob/ca19775d0ee0ce0863b46e3c76f00f3018b1e19f/test/parallel/test-stream-consumers.js#L89-L101

So blobs, should work with `Blob` objects, ArrayBuffers with `ArrayBuffer`, and so on

Reply:
Got it, will change